extensions/cache-instructions/README.md

```diff
@@ -39,11 +39,14 @@ If `CACHE_LINE_SIZE` is zero, then it _must_ be a NOP instruction.
 in which case it _may_ be 0.
 
 `NO_CACHE_START` is a cache-aligned address which represents the inclusive start of a contiguous range of physical memory addresses which will not be cached when accessed. If the privilege extension is present, this is only accessible in system mode.
-This control register is set to 0 on CPU initialization. When writing to this control register, the value will be sign extended from the current physical address width to the maxiumum supported physical address width. Writing a non-cache-aligned value
-is _unspecified_ behavior.
+This control register is set to 0 on CPU initialization.
 
 `NO_CACHE_END` is a cache-aligned address which represents the inclusive end of a contiguous range of physical memory addresses which will not be cached when accessed. If the privilege extension is present, this is only accessible in system mode.
-This control register is set to -`CACHE_LINE_SIZE` on CPU initialization. When writing to this control register, the value will be sign extended from the current physical address width to the maximum supported physical address width. Writing a
-non-cache-aligned value is _unspecified_ behavior.
+This control register is set to `-CACHE_LINE_SIZE` on CPU initialization.
 
-If `NO_CACHE_START` is larger than `NO_CACHE_END`, the non-cacheable address range wraps past the end of the address space back to the beginning.
+## Notes on `NO_CACHE_START` and `NO_CACHE_END`
+
+- The purpose of the initial values for these control registers is to intially disable caching since the location of MMIO is unknown.
+- When writing to this control register, the value will be sign extended from the current physical address width to the maxiumum supported physical address width.
```

Comment:
```suggestion
- The purpose of the initial values for these control registers is to intially disable caching since the location of MMIO is unknown.
- When writing to these control registers, the value will be sign extended from the write width to the maxiumum supported physical address width.
```

current address width => write width is unlikely to matter if people are doing reasonable things and should simplify implementation quite a bit.